wallet/src/main/scala/org/bitcoins/wallet/LockedWallet.scala

```diff
@@ -161,10 +182,186 @@ abstract class LockedWallet extends LockedWalletApi with BitcoinSLogger {
+            OutputWithIndex(out, idx)
+        }
+      }
+
+      relevantOutsWithIdx match {
+        case Nil =>
+          logger.debug(
+            s"Found no outputs relevant to us in transaction${transaction.txIdBE}")
+          Future.successful(Vector.empty)
+
+        case xs =>
+          val count = xs.length
+          val outputStr = {
+            xs.map { elem =>
+                s"${transaction.txIdBE.hex}:${elem.index}"
+              }
+              .mkString(", ")
+          }
+          logger.trace(
+            s"Found $count relevant output(s) in transaction=${transaction.txIdBE}: $outputStr")
+          if (xs.length > 1) {
+            logger.warn(
+              s"${xs.length} SPKs were relevant to transaction=${transaction.txIdBE}, but we aren't able to handle more than 1 for the time being")
+          }
+
+          def processUtxo(index: Int): Future[SpendingInfoDb] =
+            addUtxo(transaction, UInt32(index))
+              .flatMap {
+                case AddUtxoSuccess(utxo) => Future.successful(utxo)
+                case err: AddUtxoError =>
+                  logger.error(s"Could not add UTXO", err)
+                  Future.failed(err)
+              }
+
+          val addUTXOsFut: Future[Seq[(SpendingInfoDb, OutputWithIndex)]] =
+            Future
+              .sequence {
+                xs.map(out => processUtxo(out.index).map(_ -> out))
+              }
+
+          val createIncomingTxo: Seq[(SpendingInfoDb, OutputWithIndex)] => Vector[
+            IncomingWalletTXO] =
+            _.map {
+              case (utxo, OutputWithIndex(out, index)) =>
+                IncomingWalletTXO(
+                  confirmations = confirmations,
+                  // is this always the case?
+                  spent = false,
+                  scriptPubKey = out.scriptPubKey,
+                  txid = transaction.txIdBE,
+                  // always defined, as its freshly
+                  // written to the DB
+                  spendingInfoID = utxo.id.get
+                )
+            }.toVector
+
+          val incomingTXOsFut = addUTXOsFut.map(createIncomingTxo)
+          val writeIncomingTXOsFut =
+            incomingTXOsFut.flatMap(incomingTxoDAO.createAll)
+
+          writeIncomingTXOsFut
+
+      }
+    }
+  }
+
+  /**
+    * Processes an incoming transaction that already exists in our wallet.
+    * If the incoming transaction has more confirmations than what we
+    * have in the DB, we update the TX
+    */
+  private def processExistingIncomingTxo(
+      transaction: Transaction,
+      confirmations: Int,
+      foundTxo: IncomingWalletTXO): Future[Option[IncomingWalletTXO]] = {
+    if (foundTxo.confirmations < confirmations) {
+      logger.debug(
+        s"Increasing confirmation count of txo=${transaction.txIdBE}, old=${foundTxo.confirmations} new=${confirmations}")
+      val updateF =
+        incomingTxoDAO.update(foundTxo.copy(confirmations = confirmations))
```

Comment:
I don't think we should be making the assumption that this is still in the same transaction. Someone could be trying to double spend. If I understand this code correctly, we are assuming everything else is still the same about the tx and it is a simple reorg?